app/Http/Controllers/TagArticleController.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Tag;
+use Illuminate\Http\Request;
+
+class TagArticleController extends Controller
+{
+    public function index(Tag $tag)
+    {
+        return view('tagsArticles.index')
+            ->with('tag', $tag = Tag::where('id', $tag->id)->with('articles', 'articles.category')->first())
```

Comment:
Again double queries. Just use `load()`. And you don't need `->with('articles', $tag->articles);` in your blade file just use `@foreach ($tag->articles as $article) it will be much cleaner. Don't remember correctly now might be wrong but shouldn't only `with('articles.category')` work?